src/encoding.ts

```diff
@@ -422,7 +422,7 @@ export function normalizeEncoding(encoding: Encoding<string>, markDef: MarkDef):
     ) {
       if (channelDef) {
         // Array of fieldDefs for detail channel (or production rule)
-        normalizedEncoding[channel] = (isArray(channelDef) ? channelDef : [channelDef]).reduce(
+        (normalizedEncoding[channel] as any) = (isArray(channelDef) ? channelDef : [channelDef]).reduce(
```

Comment:
Why casting the LHS?

Reply:
Otherwise, we cannot assign to it. You can try to remove it but there is no type safe way to assign here. Note that previous typescript versions did this implicitly so we don't lose any type safety here.